### ISSUE TITLE
about FSQ Loss

### ISSUE THREAD

**Issue by VJJJJJJ1** [zhenye234/X-Codec-2.0#21]
hi, thank you for your work, I wonder that why the FSQ loss is not added to gen loss, as follows:

```
  if vq is True:
      # x, q, commit_loss = self.quantizer(x)
      x = x.permute(0, 2, 1)
      x, q = self.quantizer(x)
      x = x.permute(0, 2, 1)
      q = q.permute(0, 2, 1)
      return x, q, None    # why is None here?
```

edit: Sorry, I just found out that class residualFSQ does not return loss, but I wonder if the codebook can be effectively updated without loss?

**Comment by zhenye234** [comment]
fsq training does not need loss, you can refer to this paper [Finite Scalar Quantization: VQ-VAE Made Simple](https://arxiv.org/abs/2309.15505)
